lib/matplotlib/text.py

```diff
@@ -1457,7 +1459,9 @@ def _get_xy_transform(self, renderer, s):
             elif isinstance(tr, Transform):
                 return tr
             else:
-                raise RuntimeError("Unknown return type")
+                raise RuntimeError(
```

Comment:
check_isinstance could probably be adapted to handle that case as well.

Reply:
It's not worth blowing making `check_isinstance` more complex to support "Unexpected return type from callable:".

Reply:
I meant just something like `check_isinstance(types, **{"the return type of the callable": tr})` and add a cutout in check_isinstance so that `if not k.isidentifier()` then directly use `k` in the error message instead of `repr(k)` (which really just serves to add quotes around `k`, and is currently really implemented via `{!r}`).